### ISSUE TITLE
Edlib compilation error

### ISSUE THREAD

**Issue by cjw85** [nanoporetech/medaka#209]
Hello,
I tried installing the newest Medaka version to have the models for Guppy 4.0.11 and I get this error:
```
    Running setup.py install for edlib ... error
    ERROR: Command errored out with exit status 1:
     command: /home/copettid/miniconda2/envs/medaka_pip/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-5iisgb_q/edlib/setup.py'"'"'; __file__='"'"'/tmp/pip-install-5iisgb_q/edlib/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-s321c91e/install-record.txt --single-version-externally-managed --compile --install-headers /home/copettid/miniconda2/envs/medaka_pip/include/python3.7m/edlib
         cwd: /tmp/pip-install-5iisgb_q/edlib/
    Complete output (11 lines):
    running install
    running build
    running build_ext
    building 'edlib' extension
    creating build
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/edlib
    creating build/temp.linux-x86_64-3.7/edlib/src
    gcc -pthread -B /home/copettid/miniconda2/envs/medaka_pip/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Iedlib/include -I/home/copettid/miniconda2/envs/medaka_pip/include/python3.7m -c edlib.bycython.cpp -o build/temp.linux-x86_64-3.7/edlib.bycython.o -O3 -std=c++11
    gcc: error trying to exec 'cc1plus': execvp: No such file or directory
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/copettid/miniconda2/envs/medaka_pip/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-5iisgb_q/edlib/setup.py'"'"'; __file__='"'"'/tmp/pip-install-5iisgb_q/edlib/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-s321c91e/install-record.txt --single-version-externally-managed --compile --install-headers /home/copettid/miniconda2/envs/medaka_pip/include/python3.7m/edlib Check the logs for full command output.
```
this is just the final error, there were others before too.
I have python 3.7, is it a matter of the version?
Thanks!

_Originally posted by @dcopetti in https://github.com/nanoporetech/medaka/issues/181#issuecomment-703275830_

**Comment by cjw85** [comment]
@dcopetti,

This is an error in trying to compile the `edlib` dependency. It looks like you are using conda. Can you try installing the gxx_linux-64 package into your environment before installing pip?

**Comment by dcopetti** [comment]
thank you @cjw85! 
It worked and now the newest medaka runs.
If it can be useful for anybody else, these are the commands I did:
```
conda create -n medaka_pip
conda activate medaka_pip
conda install python==3.7
conda install -c anaconda gxx_linux-64 
conda install -c conda-forge pip 
pip install medaka
medaka --version
medaka 1.1.1
```

Ath the end I still got an error about pip: 
```
[...]
Building wheels for collected packages: mappy, edlib
  Building wheel for mappy (setup.py) ... done
  Created wheel for mappy: filename=mappy-2.17-cp37-cp37m-linux_x86_64.whl size=157174 sha256=58db433fea9486f6510cd7091cb8697737d6bd548904a2c9db1fedb01f8112f3
  Stored in directory: /home/copettid/.cache/pip/wheels/80/1c/d2/7cfffb95a156911cecc795784f34e31812e9fa0b82dc041d97
  Building wheel for edlib (setup.py) ... done
  Created wheel for edlib: filename=edlib-1.3.8.post1-cp37-cp37m-linux_x86_64.whl size=71747 sha256=ba5846cfed8dd358089ac76f5ffd1053c9789d805887d8907e78540e2f86ee68
  Stored in directory: /home/copettid/.cache/pip/wheels/51/74/df/a57a747f0b41fafc8748f73a5ea971ad7b17cc88e67e2bc208
Successfully built mappy edlib
Installing collected packages: pycparser, cffi, mappy, sortedcontainers, intervaltree, edlib, numpy, h5py, progressbar33, pyparsing, packaging, ont-fast5-api, chardet, idna, urllib3, requests, biopython, xopen, pyfaidx, scipy, decorator, networkx, whatshap, tensorflow-estimator, protobuf, absl-py, gast, keras-preprocessing, astunparse, opt-einsum, google-pasta, pyasn1, rsa, async-timeout, multidict, typing-extensions, yarl, attrs, aiohttp, cachetools, pyasn1-modules, google-auth, tensorboard-plugin-wit, oauthlib, requests-oauthlib, google-auth-oauthlib, werkzeug, zipp, importlib-metadata, markdown, tensorboard, termcolor, wrapt, tensorflow, parasail, medaka
ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.

tensorflow 2.2.0 requires scipy==1.4.1; python_version >= "3", but you'll have scipy 1.5.2 which is incompatible.
Successfully installed absl-py-0.10.0 aiohttp-3.6.2 astunparse-1.6.3 async-timeout-3.0.1 attrs-20.2.0 biopython-1.76 cachetools-4.1.1 cffi-1.14.3 chardet-3.0.4 decorator-4.4.2 edlib-1.3.8.post1 gast-0.3.3 google-auth-1.22.0 google-auth-oauthlib-0.4.1 google-pasta-0.2.0 h5py-2.10.0 idna-2.10 importlib-metadata-2.0.0 intervaltree-3.1.0 keras-preprocessing-1.1.2 mappy-2.17 markdown-3.2.2 medaka-1.1.1 multidict-4.7.6 networkx-2.5 numpy-1.19.2 oauthlib-3.1.0 ont-fast5-api-3.1.6 opt-einsum-3.3.0 packaging-20.4 parasail-1.2 progressbar33-2.4 protobuf-3.13.0 pyasn1-0.4.8 pyasn1-modules-0.2.8 pycparser-2.20 pyfaidx-0.5.9.1 pyparsing-2.4.7 requests-2.24.0 requests-oauthlib-1.3.0 rsa-4.6 scipy-1.5.2 sortedcontainers-2.2.2 tensorboard-2.2.2 tensorboard-plugin-wit-1.7.0 tensorflow-2.2.0 tensorflow-estimator-2.2.0 termcolor-1.1.0 typing-extensions-3.7.4.3 urllib3-1.25.10 werkzeug-1.0.1 whatshap-1.0 wrapt-1.12.1 xopen-0.9.0 yarl-1.6.0 zipp-3.3.0
```
hopefully it won't be much of a problem later on.

**Comment by cjw85** [comment]
There will shortly be a conda package available for medaka 1.1.2, so installing the PyPI package inside a conda environment should not be necessary.
